src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
```

Comment:
I think if we want to avoid heap allocation, the strategy I'd use would be to not have a planner -- have free function called `process_planless` that will recursively pull a factor out of `len` and handle it, pull another factor out of `len` and handle it, etc. This way we don't need to store FFTAlgorithm instances in the first place, so this problem goes away.